community/detectors/gitlab_cve_2021_22205/src/main/java/com/google/tsunami/plugins/detectors/cves/cve202122205/Cve202122205VulnDetector.java

```diff
@@ -0,0 +1,269 @@
+          + "6967687420225C0A22202E2071787B5453554e414d495f5343414e4e45527D202E205C0A22206220222920"
+          + "29202020202020202020202020202020202020202020202020202020202020202020202020202020202020"
+          + "20202020202020202020202020202020202020202020202020202020202020202020202020202020202020"
+          + "20202020202020202020202020202020202020202020202020202020202020202020202020202020202020"
+          + "20202020202020202020202020202020202020202020202020202020202020202020202020202020202020"
+          + "20202020202020202020202020202020202020202020202020202020202020202020202020202020202020"
+          + "20202020202020202020202020202020202020202020202020202020202020202020202020202020202020"
+          + "20202020202020202020202020202020202020202020202020202020202020202020202020202020202020"
+          + "20202020202020202020202020202020202020202020202020202020202020202020202020202020202020"
+          + "20202020202020202020202020202020202020202020202020202020202020202020202020202020202020"
+          + "20202020202020202020202020202020202020202020202020202020202020202020200A0D0A2D2D2D2D2D"
+          + "2D5765624B6974466F726D426F756E64617279494D76336D7852673539546B465358352D2D0D0A";
+
+  private final HttpClient httpClient;
+  private final Clock utcClock;
+
+  @Inject
+  Cve202122205VulnDetector(@UtcClock Clock utcClock, HttpClient httpClient) {
+    this.httpClient = checkNotNull(httpClient);
+    this.utcClock = checkNotNull(utcClock);
+  }
+
+  private static boolean isWebServiceOrUnknownService(NetworkService networkService) {
+    return networkService.getServiceName().isEmpty()
+        || NetworkServiceUtils.isWebService(networkService)
+        || NetworkServiceUtils.getServiceName(networkService).equals("unknown");
+  }
+
+  private static String buildTargetUserSignUrl(NetworkService networkService) {
+    StringBuilder targetUrlBuilder = new StringBuilder();
+    if (NetworkServiceUtils.isWebService(networkService)) {
+      targetUrlBuilder.append(NetworkServiceUtils.buildWebApplicationRootUrl(networkService));
+    } else {
+      targetUrlBuilder
+          .append("http://")
+          .append(toUriAuthority(networkService.getNetworkEndpoint()))
+          .append("/");
+    }
+    targetUrlBuilder.append(USER_SIGN_PATH);
+    return targetUrlBuilder.toString();
+  }
+
+  private static String buildTargetVulnerabilityUrl(NetworkService networkService) {
+    StringBuilder targetUrlBuilder = new StringBuilder();
+    if (NetworkServiceUtils.isWebService(networkService)) {
+      targetUrlBuilder.append(NetworkServiceUtils.buildWebApplicationRootUrl(networkService));
+    } else {
+      targetUrlBuilder
+          .append("http://")
+          .append(toUriAuthority(networkService.getNetworkEndpoint()))
+          .append("/");
+    }
+    targetUrlBuilder.append(VUL_PATH);
+    return targetUrlBuilder.toString();
+  }
```

Comment:
These two methods are very similar so they could be moved into a common function.